### DISCUSSION TITLE
Color picker position and activation

### DISCUSSION THREAD

**Discussion by WolfgangFahl** [zauberzeug/nicegui#1249] [Q&A] [answered]
### Question

color picker position is a followup issue of #1231 

But there is the followup issue how to properly position and size the picker so that it looks more pleasant.
Also i am not sure about the activation - it seems to be triggered not only by the button that opens it. I don't seem to have a proper understanding of how it is supposed to work.
http://nicescad.bitplan.com/ has the current state of affairs.

**Comment by rodja** [comment]
Sorry, @WolfgangFahl. The question is a bit vague ("properly position and size the picker so that it looks more pleasant"). Do you have some minimal executable example which you could share? What do you want to archive? How should it look? Also your deployed code will change over time which will make it hard to understand your question in 1-4 month. Please add code and maybe a screenshot.

**Comment by WolfgangFahl** [comment]
Position and size problem screen shot:
![grafik](https://github.com/zauberzeug/nicegui/assets/1336221/4b3eb5ba-dc46-4680-8241-012a41cae1e0)

**Comment by WolfgangFahl** [comment]
It's activated no only when clicking 
![grafik](https://github.com/zauberzeug/nicegui/assets/1336221/2c36360a-e165-4f05-a82e-7c79a8d305f1)
 but also when clicking anything else in the splitter
The code is:
```python
 with ui.splitter() as splitter:
                with splitter.before:
                    self.grid_button=self.tool_button("toggle grid",handler=self.toggle_grid,icon='grid_off',toggle_icon='grid_on')
                    self.axes_button=self.tool_button("toggle axes",icon="polyline",toggle_icon="square",handler=self.toggle_axes)
                    self.color_picker_button=ui.button(icon='colorize')      
                    self.color_picker = ui.color_picker(on_pick=self.pick_color)
                    self.color_picker_button.disable()

async def pick_color(self,e:ColorPickEventArguments):
        """
        Asynchronously picks a color based on provided event arguments.
    
        This function changes the color of the 'color_picker_button' and the 'stl_object'
        according to the color specified in the event arguments.
    
        Args:
            e (ColorPickEventArguments): An object containing event-specific arguments.
                The 'color' attribute of this object specifies the color to be applied.
    
        Note:
            If 'stl_object' is None, the function will only change the color of 'color_picker_button'.
            Otherwise, it changes the color of both 'color_picker_button' and 'stl_object'.
        """
        self.color_picker_button.style(f'background-color:{e.color}!important')
        if self.stl_object:
            self.stl_object.material(f'{e.color}')
        pass
                    
```
Note how the picker button doesn't even have a handler at this time.

**Comment by falkoschindler** [comment]
I just noticed that our [color picker demo](https://nicegui.io/documentation#color_picker) is wrong. The correct usage is like this:

```py
with ui.button('Pick a color'):
    ui.color_picker(on_pick=lambda: ...)
```

If you don't want to have it opened automatically when clicking into its parent container, you can set `.props('no-parent-event')`. Before NiceGUI 1.3.0 we set this prop by default. But then we removed it to support Quasar's intended way of combining buttons with menus.

**Reply by falkoschindler** [reply, marked as answer]
Fixed in [64040e7](https://github.com/zauberzeug/nicegui/commit/64040e7802568c78f140a8aacfccf271dc6b5efc).

**Comment by WolfgangFahl** [comment]
Thx - fixed via https://github.com/WolfgangFahl/nicescad/issues/24

![grafik](https://github.com/zauberzeug/nicegui/assets/1336221/f3054868-654a-4842-a800-ebfb3b16f49d)
